vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImpl.java

```diff
@@ -490,9 +518,10 @@ private MultiMap getQueryParams(Charset charset) {
     return queryParams;
   }
 
-  private Map<String, String> getPathParams() {
+  private Map<String, String> getOrCreatePathParams() {
     if (pathParams == null) {
-      pathParams = new HashMap<>();
+      // let's start small
+      pathParams = new HashMap<>(1);
```

Comment:
Isn't this too aggressive? How about using `4`, which allows to store 3 path params without rehashing?